dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -135,7 +183,7 @@ WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
       index_paths_(spec.GetRepeatedArgument<std::string>("index_paths")),
       missing_component_behavior_(detail::wds::ParseMissingExtBehavior(
           spec.GetArgument<std::string>("missing_component_behavior"))) {
-  DALI_ENFORCE(paths_.size() == index_paths_.size(),
+  DALI_ENFORCE(paths_.size() == index_paths_.size() || index_paths_.size() == 0,
```

Comment:
I would update the error message. Something like:
`"Number of webdataset archives should match the number of index files or be not provided at all");`